### PR TITLE
Disable opening Tor options in guest window

### PR DESCRIPTION
**Pull request by yrliou**
Fix https://github.com/brave/brave-browser/issues/7237
Fix https://github.com/brave/brave-browser/issues/7225

Currently we cannot open private window from a guest window, this PR is to disable opening Tor in guest window too to align with private window for consistency and also eliminate the path that would lead users to the crash described in https://github.com/brave/brave-browser/issues/7225.

Screenshots in a guest window:
Screen Shot 2019-12-06 at 8 52 37 AM
Screen Shot 2019-12-06 at 8 52 17 AM
Screen Shot 2019-12-06 at 8 52 11 AM

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
1. Open a guest window, check that there's no `New Private Window with Tor` showing up in the hamburger menu.
2. (macOS) Check that `New Private Window with Tor` in the File menu is gray out.
3. Visit brave.com in the guest window, right click on a link and check `Open Link in Private Window with Tor` is disabled in the context menu.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by yrliou** [comment]
cc @tomlowenthal for product point of view.

File: browser/ui/brave_browser_command_controller.cc [hunk 1]
```diff
@@ -146,8 +146,10 @@ void BraveBrowserCommandController::UpdateCommandForWebcompatReporter() {
 #if BUILDFLAG(ENABLE_TOR)
 void BraveBrowserCommandController::UpdateCommandForTor() {
   const bool is_tor_enabled = !tor::TorProfileService::IsTorDisabled();
+  const bool is_guest_session = browser_->profile()->IsGuestSession();
   UpdateCommandEnabled(IDC_NEW_TOR_CONNECTION_FOR_SITE, is_tor_enabled);
-  UpdateCommandEnabled(IDC_NEW_OFFTHERECORD_WINDOW_TOR, is_tor_enabled);
+  UpdateCommandEnabled(IDC_NEW_OFFTHERECORD_WINDOW_TOR,
```
Comment (simonhong) [inline browser/ui/brave_browser_command_controller.cc]: I think guest window also doesn't need `IDC_NEW_TOR_CONNECTION_FOR_SITE` command.

Comment (yrliou) [reply browser/ui/brave_browser_command_controller.cc]: Hm yes, but we currently only insert that command for Tor profile though.

Comment (simonhong) [reply browser/ui/brave_browser_command_controller.cc]: I see. it seems there is no way to trigger that command except app menu.

**Comment by yrliou** [comment]
CI failure are test-install and audit-dep failures which are unrelated to this PR.

File: chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc [hunk 2]
```diff
@@ -38,8 +38,10 @@ bool BraveRenderViewContextMenu::IsCommandIdEnabled(int id) const {
   switch (id) {
     case IDC_CONTENT_CONTEXT_OPENLINKTOR:
 #if BUILDFLAG(ENABLE_TOR)
-      if (tor::TorProfileService::IsTorDisabled())
+      if (tor::TorProfileService::IsTorDisabled() ||
```
Comment (bridiver) [inline chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc]: I thought we integrated these checks somewhere? I'd rather not have them duplicated in multiple places. Can we have `IsTorDisabledForProfile` or something?

Comment (yrliou) [reply chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc]: Addressed in https://github.com/brave/brave-core/pull/4155/commits/fd7483a6e085721e025a5970d71dea0106961d1c.

**Comment by yrliou** [comment]
building tests for mac failed with below known error, will retry macOS.
```
10:51:01  [1/6] ACTION //brave/vendor/adblock_rust_ffi:test_adblock_install_name_unit_tests(//build/toolchain/mac:clang_x64)
10:51:01  FAILED: alwaysrununittests/test_adblock_install_name 
10:51:01  python ../../build/gn_run_binary.py /usr/bin/install_name_tool -change @loader_path/../../Libraries/libadblock.dylib @executable_path/gen/adblock/out/x86_64-apple-darwin/release/libadblock.dylib brave_unit_tests
10:51:01  error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool: changing install names or rpaths can't be redone for: brave_unit_tests (for architecture x86_64) because larger updated load commands do not fit (the program must be relinked, and you may need to use -headerpad or -headerpad_max_install_names)
10:51:01  /usr/bin/install_name_tool failed with exit code 1
```

**Comment by yrliou** [comment]
> CI result: building tests for mac failed with below known error, will retry macOS; other platforms are all green.
> 
> ```
> 10:51:01  [1/6] ACTION //brave/vendor/adblock_rust_ffi:test_adblock_install_name_unit_tests(//build/toolchain/mac:clang_x64)
> 10:51:01  FAILED: alwaysrununittests/test_adblock_install_name 
> 10:51:01  python ../../build/gn_run_binary.py /usr/bin/install_name_tool -change @loader_path/../../Libraries/libadblock.dylib @executable_path/gen/adblock/out/x86_64-apple-darwin/release/libadblock.dylib brave_unit_tests
> 10:51:01  error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool: changing install names or rpaths can't be redone for: brave_unit_tests (for architecture x86_64) because larger updated load commands do not fit (the program must be relinked, and you may need to use -headerpad or -headerpad_max_install_names)
> 10:51:01  /usr/bin/install_name_tool failed with exit code 1
> ```

I've re-tried over 5 times and still cannot get around this unrelated error, but I run unit tests and browser tests locally on macOS and they're all passed expect for one known rewards browser test timeout.

**Comment by bsclifton** [comment]
@yrliou no worries about that error; it's a known issue:
https://github.com/brave/brave-browser/issues/6515

We definitely need to fix, but it will basically fail everytime unless you change something on the build and do an incremental. Restarting the build will do the whole thing again and it'll fail each time ☹️

**Pull request merged**